core/domain/html_validation_service_test.py

```diff
@@ -1672,6 +1672,123 @@ def test_validate_math_tags_in_html_with_attribute_math_content(self):
+            'alue="{&amp;quot;raw_latex&amp;quot;: &amp;quot;\\\\frac{x}{y}'
+            '&amp;quot;, &amp;quot;svg_filename&amp;quot;: &amp;quot;&amp'
+            ';quot;}"></oppia-noninteractive-math>'
+            '<oppia-noninteractive-math math_content-with-value="{&amp;'
+            'quot;raw_latex&amp;quot;: &amp;quot;+,+,+,+(x^2)&amp;quot;, &amp;'
+            'quot;svg_filename&amp;quot;: &amp;quot;&amp;quot;}"></oppia'
+            '-noninteractive-math>'
+            '<oppia-noninteractive-math math_content-with-value="{&amp;q'
+            'uot;raw_latex&amp;quot;: &amp;quot;\\\\sqrt{x}&amp;quot;, &am'
+            'p;quot;svg_filename&amp;quot;: &amp;quot;&amp;quot;}"></opp'
+            'ia-noninteractive-math>')
+
+        expected_list_of_latex = [
+            '\\sqrt{x}', '\\frac{x}{y}', '+,+,+,+(x^2)']
+        expected_size = 23000
+        expected_largest_expression = '+,+,+,+(x^2)'
+        size, largest_expression, list_of_latex = (
+            html_validation_service.
+            extract_math_rich_text_related_information_from_html(html_string2))
+        self.assertEqual(expected_size, size)
+        self.assertEqual(largest_expression, expected_largest_expression)
+        self.assertEqual(sorted(list_of_latex), sorted(expected_list_of_latex))
+
+
+
+        html_string_with_no_math = (
+            '<p><oppia-noninteractive-image filepath-with-value="abc1.png">'
+            '</oppia-noninteractive-image>Hello this is test case to check that'
+            ' dimensions are added to the oppia noninteractive image tags.</p>'
+        )
+        self.assertEqual(
+            html_validation_service.
+            extract_math_rich_text_related_information_from_html(
+                html_string_with_no_math), (0, '', []))
+
+    def test_validate_svg_image_filenames_in_math_rte_components(self):
+        """Test that the validate_svg_image_filenames_in_math_rte_components
+        method checks all the filenames in the math-tag to have images saved in
+        gcs.
+        """
+        html_string_with_filename_having_image_saved = (
+            '<p>Feedback</p><oppia-noninteractive-math math_content-with-v'
+            'alue="{&amp;quot;raw_latex&amp;quot;: &amp;quot;+,-,-,+'
+            '&amp;quot;, &amp;quot;svg_filename&amp;quot;: &amp;quot'
+            ';img.png&amp;quot;}"></oppia-noninteractive-math>'
+        )
+        with python_utils.open_file(
+            os.path.join(feconf.TESTS_DATA_DIR, 'img.png'), 'rb',
+            encoding=None) as f:
+            raw_image = f.read()
+        fs = fs_domain.AbstractFileSystem(
+            fs_domain.GcsFileSystem(
+                feconf.ENTITY_TYPE_EXPLORATION, 'exp_id_1'))
+        fs.commit('image/img.png', raw_image, mimetype='image/png')
+        self.assertEqual(
+            len(
+                html_validation_service.
+                validate_svg_image_filenames_in_math_rte_components(
```

Comment:
Do not assert on the length. Compare the return value with the actual value.

In general try to use the strictest test possible. Ditto elsewhere.

Reply:
done